tools/fs/FS_Dump_Parser/src/smartfs_dump_parser/data_model/ModelProvider.java

```diff
@@ -0,0 +1,101 @@
+package smartfs_dump_parser.data_model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public enum ModelProvider {
+	INSTANCE;
+
+	private static final int SECTOR_SIZE = 512;
+	private static final int SECTOR_HEADER_SIZE = 5;
+	private static final int CHAIN_HEADER_SIZE = 5;
+	private static final int ENTRY_HEADER_SIZE = 8;
+	private static final int FILE_NAME_LENGTH = 32;
+	private static final int ACTIVE_SECTOR_STATUS_VALUE = 101;
+	private static final int ROOT_LOGICAL_SECTOR_ID = 3;
+	private static final String SMARTFS_ROOT = "SMARTFS ROOT";
+
+	private static List<Sector> sectorList = new ArrayList<Sector>();
+	private List<Sector> activeSectors;
```

Comment:
ok you already defined 3 type of list, then you can fill each type sector's list during the addSector() working.

Reply:
Okay, modified.